dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -123,4 +127,126 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
+        ));
+
+        maxSubKeyLen++; // Account for the null character
+
+        // Loop through registered ComServerGuids, grab the ServerId and check the registered Arguments/Executable
+        for (DWORD i = 0; i < comServerGuids; i++)
+        {
+            std::vector<wchar_t> comServerGuid;
+            comServerGuid.resize(maxSubKeyLen);
+            DWORD cbName{ maxSubKeyLen };
+            THROW_IF_FAILED(RegEnumKeyExW(hKey.get(), i, comServerGuid.data(), &cbName, nullptr, nullptr, nullptr, nullptr));
+            // Path: HKLM\SOFTWARE\Classes\PackagedCom\Package\{PackageFullName}\Class\{comServerGuid}
+            std::wstring clsidPathWithKey{ clsidPath + L"\\" + comServerGuid.data() };
+
+            // Get the ServerId for the ComActivatorGuid
+            DWORD serverId = 0;
+            DWORD serverIdSize = sizeof(serverId);
+            THROW_IF_FAILED(RegGetValueW(
+                HKEY_LOCAL_MACHINE,
+                clsidPathWithKey.c_str(),
+                L"ServerId",
+                RRF_RT_REG_DWORD,
+                nullptr /* pdwType */,
+                &serverId,
+                &serverIdSize));
+
+            // Path: HKLM\SOFTWARE\Classes\PackagedCom\Package\{PackageFullName}\Server\{serverId}
+            std::wstring serverPath{ packagedRegistryPath + LR"(\Server\)" + std::to_wstring(serverId) };
+            WCHAR argumentsBuffer[MAX_VALUE_NAME];
+            DWORD argumentsBufferLength = sizeof(argumentsBuffer);
+            THROW_IF_FAILED(RegGetValueW(
+                HKEY_LOCAL_MACHINE,
+                serverPath.c_str(),
+                L"Arguments",
+                RRF_RT_REG_SZ,
+                nullptr /* pdwType */,
+                &argumentsBuffer,
+                &argumentsBufferLength));
+
+            std::wstring argumentString{ argumentsBuffer };
+
+            WCHAR exeBuffer[MAX_VALUE_NAME];
+            DWORD exeBufferLength = sizeof(exeBuffer);
+            THROW_IF_FAILED(RegGetValueW(
+                HKEY_LOCAL_MACHINE,
+                serverPath.c_str(),
+                L"Executable",
+                RRF_RT_REG_SZ,
+                nullptr /* pdwType */,
+                &exeBuffer,
+                &exeBufferLength));
+
+            std::wstring exeString{ exeBuffer };
+            THROW_IF_FAILED(GetExeNameFromPath(exeString));
+
+            wil::unique_cotaskmem_string exePath;
+            THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, exePath));
+
+            std::wstring exeToCheck{ exePath.get() };
+            THROW_IF_FAILED(GetExeNameFromPath(exeToCheck));
+
+            if (argumentString == argumentToCheck && exeString == exeToCheck)
+            {
+                std::wstring storedComActivatorString{ comServerGuid.data() };
+                comServerClsid = winrt::guid(storedComActivatorString.substr(1, storedComActivatorString.size() - 2));
+                return S_OK;
+            }
+        }
+        return E_NOT_SET;
```

Comment:
End the API call like this to make things clearer:

THROW_HR_IF(E_NOT_SET, found != true);
return S_OK;